argon2.js

```diff
@@ -81,13 +81,19 @@ const needsRehash = (digest, options) => {
 };
 
 const verify = async (digest, plain, options) => {
+  const obj = deserialize(digest);
+  // Only these have the "params" key, so if the password was encoded
+  // using any other method, the destructuring throws an error
+  if (!['argon2i', 'argon2d', 'argon2id'].includes(obj.id)) {
```

Comment:
You could use `if (!(obj.id in types)`, `types` is defined as `{ argon2d: 0, argon2i: 1, argon2id: 2 }` and avoids duplicating data.

Reply:
```suggestion
  if (!(obj.id in types)) {
```